pkg/process/metadata/parser/java/util.go

```diff
@@ -0,0 +1,146 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package javaparser
+
+import (
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/rickar/props"
+	"github.com/vibrantbyte/go-antpath/antpath"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	// maxParseFileSize is the maximum file size in bytes the parser will accept.
+	maxParseFileSize = 1024 * 1024
+)
+
+// mapSource is a type holding properties stored as map. It implements PropertyGetter
+type mapSource struct {
+	m map[string]string
+}
+
+// Get the value for the supplied key
+func (y *mapSource) Get(key string) (string, bool) {
+	val, ok := y.m[key]
+	return val, ok
+}
+
+// GetDefault gets the value for the supplied key or the defVal if missing
+func (y *mapSource) GetDefault(key string, defVal string) string {
+	val, ok := y.m[key]
+	if !ok {
+		return defVal
+	}
+	return val
+}
+
+// newArgumentSource a PropertyGetter that is taking key=value from the list of arguments provided
+// it can be done to parse both java system properties (the prefix is `-D`) or spring boot property args (the prefix is `--`)
+func newArgumentSource(arguments []string, prefix string) *mapSource {
+	parsed := make(map[string]string)
+	for _, val := range arguments {
+		if !strings.HasPrefix(val, prefix) {
+			continue
+		}
+		parts := strings.SplitN(val[len(prefix):], "=", 2)
+		if len(parts) == 1 {
+			parsed[parts[0]] = ""
+		} else {
+			parsed[parts[0]] = parts[1]
+		}
+	}
+	return &mapSource{parsed}
+}
+
+// newPropertySourceFromStream create a PropertyGetter by selecting the most appropriate parser giving the file extension.
+// An error will be returned if the filesize is greater than maxParseFileSize
+func newPropertySourceFromStream(rc io.Reader, filename string, filesize uint64) (*props.PropertyGetter, error) {
+	if filesize > maxParseFileSize {
+		return nil, fmt.Errorf("unable to parse %q. max file size exceeded(actual: %d, max: %d)", filename, filesize, maxParseFileSize)
+	}
+	var properties props.PropertyGetter
+	var err error
+	ext := strings.ToLower(filepath.Ext(filename))
+	switch ext {
+	case ".properties":
+		properties, err = props.Read(rc)
+	case ".yaml", ".yml":
+		properties, err = newYamlSource(rc)
+	default:
+		return nil, fmt.Errorf("unhandled file type for %q", filename)
+	}
+	return &properties, err
+}
+
+// longestPathPrefix extracts the longest path's portion that's not a pattern (i.e. /test/**/*.xml will return /test/)
+func longestPathPrefix(pattern string) string {
+	idx := strings.IndexAny(pattern, "?*")
+	if idx < 0 {
+		return pattern
+	}
+	idx = strings.LastIndex(pattern[:idx], "/")
```

Comment:
is the code intended to run only on linux?
if not, then it will always fail on windows

Reply:
It's an ant pattern hence using URIs (always `/`). Path are translated to URI before using `fromSlash` to cope with `\` used on windows. I have tests and looks working as expected